etna/models/seasonal_ma.py

```diff
@@ -128,18 +145,21 @@ def forecast(self, ts: TSDataset, prediction_size: int, return_components: bool
         ValueError:
             if forecast context contains NaNs
         """
-        if return_components:
-            raise NotImplementedError("This mode isn't currently implemented!")
-
         df = ts.to_pandas()
         new_df = self._forecast(df=df, prediction_size=prediction_size)
         ts.df = new_df
+
+        if return_components:
+            df.loc[-prediction_size:, pd.IndexSlice[:, "target"]] = new_df.loc[:, pd.IndexSlice[:, "target"]]
```

Comment:
Maybe just do prediction inplace in `df` and drop this assignment?

Reply:
Don't get the idea)